### PR TITLE
Add custom error pages.

### LINKED ISSUE #3: Add 404 page
the 404 page was previously tracked in GA

https://github.com/maxbeatty/jsperf.com/blob/breakdown/_tpl/status404.tpl

### LINKED ISSUE #4: Add 403 page
previously tracked by GA

https://github.com/maxbeatty/jsperf.com/blob/breakdown/_tpl/status403.tpl

### LINKED ISSUE #5: Add 400 page
previously tracked by GA

https://github.com/maxbeatty/jsperf.com/blob/breakdown/_tpl/status400.tpl

### PR DESCRIPTION
**Pull request by asilluron**
Closes #5
Closes #4
Closes #3

### CODE REVIEW

File: server/web/errors/index.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    switch(statusCode){
+      case 400:
+        return reply.view("errors/400").code(400);
+      case 403:
+        return reply.view("errors/403").code(403);
+      case 404:
+        return reply.view("errors/404").code(404);
+      default: 
+        return reply.view("errors/general");
```
Comment (maxbeatty) [inline server/web/errors/index.js]: should `code` be set here to whatever `statusCode` is here? along that line, could determine view based on status and then return once

```javascript
var viewPath = "general";
switch(statusCode) {
  case 400:
    viewPath = statusCode;
  case 403:
    viewPath = statusCode;
  case 404:
    viewPath = statusCode
  // future viewPaths may not coincide directly w/ `statusCode`
}

return reply.view("errors/" + viewPath).code(statusCode);
```

food for thought

File: server/web/errors/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  server.ext("onPreResponse", function (request, reply) {
+
+    var response = request.response;
+    if (!response.isBoom) {
+        return reply.continue();
+    }
+
+    // Replace error with friendly HTML
+    var error = response;
```
Comment (maxbeatty) [inline server/web/errors/index.js]: nitpick but looks like 

1. `request.response` becomes `response` 
2. `response` is used once 
3. `response` becomes `error`
4. `error` is used once

I'm all for shortening the call paths just seems like could do it with a few less vars. Would enjoy hearing how you see it.

```javascript
if (!request.response.isBoom) {
  return reply.continue();
}

var statusCode = request.response.output.statusCode;
```

Comment (asilluron) [inline server/web/errors/index.js]: I am using the var names as a hint of what has been inferred as the response is being checked. 

So in other words, once we know ``` !request.response.isBoom ``` that fails, then we are dealing with an error, but before that we were dealing with a generic response.

Comment (maxbeatty) [inline server/web/errors/index.js]: that makes sense. thanks for the explanation :neckbeard:

**Comment by asilluron** [comment]
@maxbeatty I took some of the comments and made the route handler much shorter and generic. In the future we can probably use a redis-backed cached glob call to add/remove error pages.

**Pull request merged**